dali/operators/math/expressions/expression_factory_instances/expression_impl_factory.h

```diff
@@ -181,8 +171,8 @@ std::unique_ptr<ExprImplBase> ExprImplFactoryTernaryOp(const ExprFunc &expr) {
 #define IMPLEMENT_OP_FACTORY_GPU_BINARY(OP)                                                 \
   std::unique_ptr<ExprImplBase> OpFactory(arithm_meta<ArithmeticOp::OP, GPUBackend>,        \
                                           const ExprFunc &expr) {                           \
-    return ExprImplFactoryBinOp<ArithmeticOp::OP, GPUBackend, ExprImplGpuTT, ExprImplGpuTC, \
-                                ExprImplGpuCT>(expr);                                       \
+    return ExprImplFactoryBinOp<ArithmeticOp::OP, GPUBackend, ExprImplGpuTT, ExprImplGpuTT, \
```

Comment:
Now there is a single implementation for the three cases (I am not removing the logic to select the 3 implementations because it is used for the CPU implementation)